data_steward/retraction/participant_row_counts.py

```diff
@@ -0,0 +1,248 @@
+    :return: df with headers table_id, all_counts, all_ehr_counts, and map_ehr_counts
+    """
+    dataset_type = ru.get_dataset_type(dataset_id)
+    counts_df = pd.DataFrame(columns=[
+        ru_consts.TABLE_ID, consts.ALL_COUNT, consts.ALL_EHR_COUNT,
+        consts.MAP_EHR_COUNT
+    ])
+    table_df = bq.get_table_info_for_dataset(project_id, dataset_id)
+
+    if dataset_type == common.COMBINED:
+        query = get_combined_deid_query(project_id, dataset_id, pid_source,
+                                        table_df)
+    elif dataset_type == common.DEID or dataset_type == common.RELEASE:
+        query = get_combined_deid_query(project_id,
+                                        dataset_id,
+                                        pid_source,
+                                        table_df,
+                                        for_deid=True)
+    elif dataset_type == common.EHR:
+        query = get_ehr_query(project_id, dataset_id, pid_source, hpo_id,
+                              table_df)
+    elif dataset_type == common.RDR:
+        query = get_dataset_query(project_id,
+                                  dataset_id,
+                                  pid_source,
+                                  table_df,
+                                  for_rdr=True)
+    else:
+        query = get_dataset_query(project_id, dataset_id, pid_source, table_df)
+
+    if query:
+        counts_df = bq.query(query, project_id)
+        # ignore zero counts
+        counts_df = counts_df[counts_df[consts.ALL_COUNT] > 0]
+    return counts_df
+
+
+def log_total_rows(df, dataset_id):
+    """
+    Logs rows from counts dataframe
+
+    :param df: dataframe containing counts
+    :param dataset_id: Identifies the dataset under consideration
+    :return:
+    """
+    rows = df.get_values()
+    if rows.size > 0:
+        for count_row in rows:
+            logging.info('{}, {}, {}, {}, {}'.format(dataset_id, *count_row))
+
+
+def count_pid_rows_in_project(project_id, hpo_id, pid_source, dataset_ids=None):
+    """
+    Logs dataset_name, table_id, all_count, all_ehr_count and map_ehr_count to count rows pertaining to pids
+
+    :param project_id: identifies the project
+    :param hpo_id: Identifies the hpo site that submitted the pids
+    :param pid_source: string containing query or list containing pids
+    :param dataset_ids: list identifying datasets to retract from or None to retract from all datasets
+    :return:
+    """
+    dataset_ids = ru.get_dataset_ids_to_target(project_id, dataset_ids)
+    for dataset_id in dataset_ids:
+        try:
+            # We do not fetch queries for each dataset here and union them since it exceeds BQ query length limits
+            counts_df = count_pid_rows_in_dataset(project_id, dataset_id,
+                                                  hpo_id, pid_source)
+            log_total_rows(counts_df, dataset_id)
+        except BadRequest:
```

Comment:
This obscures an error which we programmers likely have caused (it's a 400 bad client request error). Is it possibly better for the function to just fail? Or is this is a stopgap to account for a lot of non-conforming datasets which we expect would fail when this is run against all datasets in the project? If the latter and the program should continue, could the `BadRequest` exception be captured and logged here?

Reply:
Yes, it is a stopgap. There are instances where in older datasets, the `src_hpo_id/src_id` does not exist in mapping tables and can cause errors. The `logging.exception` should provide the stacktrace, but let me know if I can add any improvements to it.

Reply:
Sorry, this is my mistake: I did not know `logging.exception` derives the full trace. So this actually provides **additional** information than I thought. Thanks!